static/src/javascripts/bootstraps/enhanced/football.js

```diff
@@ -0,0 +1,426 @@
+// @flow
+import bean from 'bean';
+import bonzo from 'bonzo';
+import $ from 'lib/$';
+import config from 'lib/config';
+import fetchJson from 'lib/fetch-json';
+import {
+    belowArticleVisible,
+    isCompetition,
+    isMatch,
+    isFootballStatsPage,
+    isLiveClockwatch,
+} from 'lib/page';
+import reportError from 'lib/report-error';
+import { TableDoughnut } from 'common/modules/charts/table-doughnut';
+import {
+    statsFor,
+    tableFor,
+    matchDayFor,
+} from 'common/modules/sport/football/football';
+import { MatchInfo } from 'common/modules/sport/football/match-info';
+import { MatchListLive } from 'common/modules/sport/football/match-list-live';
+import { tagPageStats } from 'common/modules/sport/football/tag-page-stats';
+import { ScoreBoard } from 'common/modules/sport/score-board';
+import { addComponent } from 'common/modules/ui/rhc';
+import { replaceLocaleTimestamps } from 'common/modules/ui/relativedates';
+
+declare type Extras = Array<?{
+    content?: Element,
+    importance?: number,
+    name?: string,
+    ready: boolean,
+}>;
```

Comment:
Would it make more sense to define `Extra` as a type, rather than `Extras`? There's a whole lotta `if(extra ...)` checks later on that I presume come from the maybe-ness of this array's content